api/envoy/config/core/v3/base.proto

```diff
@@ -347,7 +347,13 @@ message HeaderValueOption {
   // Describes the supported actions types for header append action.
   enum HeaderAppendAction {
     // This action will append the specified value to the existing values if the header
-    // already exists. If the header doesn't exist then this will add the header with
+    // already exists and the header is inline header.
```

Comment:
Let's revert this change in API proto.

`APPEND_IF_EXISTS_OR_ADD ` just need to follow Envoy's `HeaderMapImpl::addCopy` and we don't need to impose additional restriction on it.

Reply:
No problem at all, I will revert it back.